### PR TITLE
Draft for promises

### PR DESCRIPTION
**Pull request by karan-pathak**
#2264

### CODE REVIEW

**Comment by jenweber** [comment]
Hi @karan-pathak ! This is awesome 😄 I see a few typos and small grammar things - is it ok if I add a commit to your PR, or would you rather that I send them to you and you can make the changes?

File: source/getting-started/js-primer.md [hunk 1]
```diff
@@ -147,10 +147,76 @@ console.log(i) // ReferenceError: i is not defined
...
+
+In the above snippet, `store.findRecord('person', 1)` can make a network request if the data is not
+already present in the store. Thus it returns a `promise` object which can resolve almost instantly if the data is present in store or it can take some time to resolve if the data is being fetched by a network request.
+
+Now we can come to part where these promises are chained:
+
+```javascript
+store.findRecord('person', 1).then(function(person) {
+
+  store.findRecord('post', 1) //query for another record.
```
Comment (josemarluedke) [inline source/getting-started/js-primer.md]: If I'm understanding correctly this example, shound't this line have `return` so the next `then` would receive the returned value, in this case `post`?

**Comment by karan-pathak** [comment]
Hi @jenweber, Thanks for reviewing it! please feel free to make changes to this PR, and if you want me to make the changes then just let me know :)

**Comment by jenweber** [comment]
@karan-pathak take a look and let me know what you think of my edits!

After looking at this more closely, I think it needs 2 more points:
- a sentence or two about how variables modified inside a promise will be `undefined` for non-Promise-wrapped code (I don't know how to say this better. Does that make any sense to you?)
- A closing statement

I am also going to ping @locks and/or @jaredgalanis for review. Promises are hard so some more feedback would be good.

**Comment by karan-pathak** [comment]
@jenweber, great edits! The extra points that you suggested seem valid. I'll add them to this PR. 

Also, more feedback is always welcome :)

**Comment by toddjordan** [comment]
Thanks for contributing @karan-pathak !

**Pull request merged**